.github/workflows/actions/setup_dev_ubuntu/install_libraries

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+set -x
+
+# Install ghc and cabal
+ghcup install ghc 8.10.7
+ghcup set ghc 8.10.7
+
+mkdir -p src
+mkdir -p srv/bin
+# Install libsodium
+(
+  cd src
+  git clone https://github.com/input-output-hk/libsodium
+  cd libsodium
+  git checkout 66f017f1
+  ./autogen.sh
+  ./configure --prefix="$(realpath $PWD/../../srv)"
+  make
+  make install
+)
+
+# Install secp256k1
+(
+  cd src
+  git clone https://github.com/bitcoin-core/secp256k1
+  cd secp256k1
+  git checkout ac83be33
+  ./autogen.sh
+  ./configure --enable-module-schnorrsig --enable-experimental --prefix="$(realpath $PWD/../../srv)"
+  make
+  make install
+)
+
+# Install cardano
+(
+  cd src
+  curl https://update-cardano-mainnet.iohk.io/cardano-node-releases/cardano-node-1.35.7-linux.tar.gz -o cardano-node.tar.gz
+  cd ../srv/bin
+  tar xfz ../../src/cardano-node.tar.gz
+)
+
+cat <<EOF
+# Setup the following environment varialbles:
+LD_LIBRARY_PATH="$(realpath $PWD/srv)/lib:${LD_LIBRARY_PATH}"
+PKG_CONFIG_PATH="$(realpath $PWD/srv)/lib/pkgconfig:${PKG_CONFIG_PATH}"
+PATH="$(realpath $PWD/srv)/bin:${PATH}
```

Comment:
If we would be using the user's home `~/.local`, maybe these additional env variables would not be needed?